airgun/widgets.py

```diff
@@ -2097,11 +2097,30 @@ def fill(self, values):
 
 class AutoCompleteTextInput(TextInput):
     """Autocomplete Search input field, We must remove the focus from this widget after fill to
-    force the auto-completion list to be hidden.
+    force the auto-completion list to be hidden. Since this is a react component, calling browser
+    clear method directly on the field has no effect, thus we need to clear the field using the 
+    clear button attached to the input
     """
 
+    clear_button = Text(
+        locator="//span[contains(@class,'autocomplete-clear-button') or "
+                "contains(@class,'fa-close')]"
+    )
+
+    def clear(self):
+        """Clears search field value and re-trigger search to remove all
+        filters.
+        """
+        if self.clear_button.is_displayed:
+            self.clear_button.click()
+
     def fill(self, value):
-        changes = super().fill(value)
+        if self.value != value:
+            changes = True
+        else:
+            changes = False
```

Comment:
```python
changes = self.value != value
```
?

I see in the past we assigned return value of `super().fill()` to changes, so probably something like below would be more in line with that intent?
```python
def fill(self, value):
    old_value = self.value
    self.clear()
    super().fill(value)
    self.browser.plugin.ensure_page_safe()
    self.browser.execute_script('arguments[0].blur();', self.__element__())
    return self.value != old_value
```